apps/solution/src/components/people/people.tsx

```diff
@@ -1,18 +1,47 @@
 import { Grid } from '@libs/design';
+import { Dispatch, SetStateAction, useState } from 'react';
 import Person from '../person/person';
 import usePeople from '../person/usePeople';
 
 import style from './people.module.scss';
 
+function normalize(text: string) {
+  return text.toLowerCase().normalize('NFD').replace(/[\u0300-\u036f]/g, "")
+}
+
 export function People() {
   const people = usePeople();
+  const [searchTerm, setSearchTerm] = useState("")
+
+  const filteredPeople = people?.filter(person => {
```

Comment:
This probably can be wrapped in a `useMemo` reacting on `people` and `searchTerm` right ?

Reply:
Indeed a good way to teach the importance of memoization in cases like that to increase performance

Reply:
Isn't it better to just "chain" the hooks in the main people.jsx and show that it reacts on changes if you use a useMemo inside the usePeopleFilter(people)

Such as

```
const people = usePeople();
const peopleFiltered = usePeopleFilter(people, searchTerm); // <- useMemo(() => {}, [people, searchTerm])
````

Reply:
I think both ways have they benefits. I would be totally aggree with you solution if `usePeopleFilter` could be generic enough ; but we can't make this a generic `useFilter` as there is some logic inside  `usePeopleFilter` (the fact that we use `firstname` and `lastname` for filtering, for example). 
In the other hand, `usePeopleFilter` depends on `usePeople`, so they're linked, one way or another. 
That said, your solution seems totally correct to me, so you can keep it as is, I think it's simpler.